`script` crate in `master` does not compile with `nightly`

**Describe the bug:**

While `cargo build`ing a project that has some `servo` GH deps, I get this:

```
error[E0283]: type annotations needed
    --> /home/ritz/.cargo/git/checkouts/servo-e53a6e7b994a25fe/e9ed5dd/components/script/script_thread.rs:1729:84
     |
1729 |                 if pipeline_id == doc_id && task_duration.as_nanos() > MAX_TASK_NS.into() {
     |                                                                      -             ^^^^
     |                                                                      |
     |                                                                      type must be known at this point
     |
     = note: multiple `impl`s satisfying `u128: PartialOrd<_>` found in the following crates: `core`, `deranged`:
             - impl PartialOrd for u128;
             - impl<MIN, MAX> PartialOrd<deranged::RangedU128<MIN, MAX>> for u128
               where the constant `MIN` has type `u128`, the constant `MAX` has type `u128`;
help: try using a fully qualified path to specify the expected types
     |
1729 -                 if pipeline_id == doc_id && task_duration.as_nanos() > MAX_TASK_NS.into() {
1729 +                 if pipeline_id == doc_id && task_duration.as_nanos() > <u64 as std::convert::Into<T>>::into(MAX_TASK_NS) {
     |

For more information about this error, try `rustc --explain E0283`.
error: could not compile `script` (lib) due to 1 previous error
```

**To Reproduce:**

This is my Cargo.toml:
```toml
[package]
name = "servo-renderer"
version = "0.1.0"
edition = "2021"

[dependencies]
# Core Servo components
servo_config = { git = "https://github.com/servo/servo" }
servo_geometry = { git = "https://github.com/servo/servo.git"}
servo_url = { git = "https://github.com/servo/servo.git" }
servoshell = { git = "https://github.com/servo/servo.git" }
webrender = { git = "https://github.com/servo/webrender" }
euclid = "0.22"
surfman = "0.9"

# Image processing
image = "0.25"
exr = "1.73"
rayon = "1"
lcms2 = "6"
```

**Platform:**
Ubuntu 24.10, `rustc --version`:

```
rustc 1.87.0-nightly (aa8f0fd71 2025-03-23)
```

--- a/components/metrics/lib.rs
+++ b/components/metrics/lib.rs
@@ -18,7 +18,7 @@ use servo_url::ServoUrl;
 
 /// TODO make this configurable
 /// maximum task time is 50ms (in ns)
-pub const MAX_TASK_NS: u64 = 50000000;
+pub const MAX_TASK_NS: u128 = 50000000;
 /// 10 second window
 const INTERACTIVE_WINDOW_SECONDS: Duration = Duration::from_secs(10);
 

--- a/components/script/script_thread.rs
+++ b/components/script/script_thread.rs
@@ -1726,7 +1726,7 @@ impl ScriptThread {
         let task_duration = start.elapsed();
         for (doc_id, doc) in self.documents.borrow().iter() {
             if let Some(pipeline_id) = pipeline_id {
-                if pipeline_id == doc_id && task_duration.as_nanos() > MAX_TASK_NS.into() {
+                if pipeline_id == doc_id && task_duration.as_nanos() > MAX_TASK_NS {
                     if self.print_pwm {
                         println!(
                             "Task took longer than max allowed ({:?}) {:?}",

